### PR TITLE
Instrument Code & Add Cypress Code Coverage Report to Github Actions/CI Workflow

### PR DESCRIPTION
**Pull request by rjohnson2011**
## Description
Run the Cypress Code Coverage report generator after instrumenting the E2E Cypress tests within the `/src` directory of the CI workflow. Upon creating a successful code coverage summary, upload this artifact via Github Actions for future ingestion into BigQuery.

## Original issue(s)
[Zenhub Ticket](https://app.zenhub.com/workspaces/planning-board-qa-standards-612411db8ec47700127369e2/issues/department-of-veterans-affairs/va.gov-team/37766)

## Testing done
Successfully ran through the CI workflow and generated a code coverage summary, and uploaded this artifact via Github Actions.

## Screenshots
Instrumenting code coverage on CI:
image

Uploaded Artifact on CI:
image

## Acceptance criteria
- [x] CI workflow successfully instruments the /src application directory.
- [x] CI workflow retrieves a summary report.
- [x] Summary report is uploaded via GHA into an artifact.
- [x] Code coverage artifact is successfully downloaded.
- [x] All CI checks are passing.

### CODE REVIEW

**Review by timwright12 — APPROVED** [review]

File: src/platform/testing/e2e/cypress/plugins/index.js [hunk 1]
```diff
@@ -1,8 +1,8 @@
 const fs = require('fs');
 const webpackPreprocessor = require('@cypress/webpack-preprocessor');
-const table = require('table').table;
-const DefinePlugin = require('webpack').DefinePlugin;
-const ProvidePlugin = require('webpack').ProvidePlugin;
+const { table } = require('table');
+const { DefinePlugin } = require('webpack');
+const { ProvidePlugin } = require('webpack');
```
Comment (holdenhinkle) [inline src/platform/testing/e2e/cypress/plugins/index.js:5]: I think you can destructure `DefinePlugin` and `ProvidePlugin` on the same line:

`const { DefinePlugin, ProvidePlugin} = require('webpack');`

File: src/platform/testing/e2e/cypress/plugins/index.js [hunk 2]
```diff
@@ -75,4 +78,5 @@ module.exports = async on => {
     table: message => console.log(table(message, tableConfig)) || null,
     /* eslint-enable no-console */
   });
+  return config;
```
Comment (holdenhinkle) [inline src/platform/testing/e2e/cypress/plugins/index.js:81]: Maybe put a nice lil empty line before the `return` statement?

**Review by holdenhinkle — APPROVED** [review]
Just a couple of small comments.

Approved.

**Review by timwright12 — APPROVED** [review]

File: src/platform/testing/e2e/cypress/plugins/index.js (hunk 1 above)
Comment (rjohnson2011) [reply src/platform/testing/e2e/cypress/plugins/index.js:5]: Good idea! Put this change in.

File: src/platform/testing/e2e/cypress/plugins/index.js (hunk 2 above)
Comment (rjohnson2011) [reply src/platform/testing/e2e/cypress/plugins/index.js:81]: Sure thing!

**Review by CBonade — APPROVED** [review]

File: .babelrc [hunk 3]
```diff
@@ -40,6 +40,7 @@
   ],
   "plugins": [
     "lodash",
+    "istanbul",
```
Comment (ddzz) [inline .babelrc:43]: Can istanbul be moved to the `test` environment?

File: .github/workflows/continuous-integration.yml [hunk 4]
```diff
@@ -489,6 +492,16 @@ jobs:
           TESTS: ${{ needs.cypress-tests-prep.outputs.tests }}
           APP_URLS: ${{ needs.cypress-tests-prep.outputs.app_urls }}
 
+      - name: View code coverage summary
+        run: npx nyc report --reporter=text-summary
+
+      - name: Create coverage artifact
+        uses: actions/upload-artifact@v2
```
Comment (ddzz) [inline .github/workflows/continuous-integration.yml:499]:
```suggestion
        uses: actions/upload-artifact@v3
```

File: .babelrc (hunk 3 above)
Comment (rjohnson2011) [reply .babelrc:43]: I tried placing istanbul in the `test` environment but then the code coverage report would not generate during the GHA workflow.

**Pull request merged**